### PR TITLE
[SPARK-14460] [SQL] properly handling of column name contains space

### PR DESCRIPTION
**Pull request by bomeng**
## What changes were proposed in this pull request?

Although it is not recommended, table can be created with column name containing space. For example, 

> create table test.people1 (name TEXT(32) NOT NULL,`the id` INTEGER NOT NULL)

When we query the table via JDBC, we also need to quote the column name in the same way if the column name contains space.

I am leveraging the quote method in the dialect, it will add the proper quote to the column name based on its data source. 
## How was this patch tested?

I've updated the test case to test this scenario.

### CODE REVIEW

**Comment by bomeng** [comment]
@andrewor14 Could you please review my codes?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala [hunk 1]
```diff
@@ -246,13 +247,23 @@ object JdbcUtils extends Logging {
...
+    dialect.quoteIdentifier(columnName)
+  }
+
+  /**
    * Compute the schema string for this RDD.
    */
-  def schemaString(df: DataFrame, url: String): String = {
+  def schemaString(dialect: JdbcDialect, df: DataFrame, url: String): String = {
     val sb = new StringBuilder()
     val dialect = JdbcDialects.get(url)
```
Comment (ash211) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala]: Is the new dialect you're passing in different from this one in some way?

Comment (bomeng) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala]: The purpose to pass in dialect is to get proper quote for columns based on its data source. Any suggestion?

Comment (ash211) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala]: You pass in a parameter named dialect to the schemaString method, but there's also the dialect that comes from JdbcDialects.get(url) --- that's the duplicate I was trying to point out

Comment (bomeng) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala]: Thanks for pointing out. I've modified the codes. Please check it out.

**Comment by cloud-fan** [comment]
seems it's fixed in https://github.com/apache/spark/pull/15662 ?

**Comment by HyukjinKwon** [comment]
ping @bomeng for ^.

**Pull request closed without merging**